CMakeLists.txt

```diff
@@ -102,6 +106,64 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Generic")
     if(PLATFORM_NAME STREQUAL "nuttx")
         set(PLATFORM_NAME_NUTTX ON)
+        set(TRANSPORT_TYPES "UDP" "TCP" "SERIAL")
+    endif()
+endif()
+
+# Check external transport.
+SET(PLATFORM_EXTERNAL_TRANSPORT_HEADER "" CACHE PATH "External transports header file")
+SET(PLATFORM_EXTERNAL_TRANSPORT "" CACHE PATH "External transports code file")
+
+if( PLATFORM_EXTERNAL_TRANSPORT)
+    set(PLATFORM_EXTERNAL ON)
+
+    set(TRANSPORT_TYPES "SERIAL")
+
+    get_filename_component(PLATFORM_EXTERNAL_TRANSPORT_HEADER_FILENAME ${PLATFORM_EXTERNAL_TRANSPORT_HEADER} NAME)
+    get_filename_component(PLATFORM_EXTERNAL_TRANSPORT_FILENAME ${PLATFORM_EXTERNAL_TRANSPORT} NAME)
+    get_filename_component(PLATFORM_EXTERNAL_TRANSPORT_HEADER_DIRECTORY ${PLATFORM_EXTERNAL_TRANSPORT_HEADER} DIRECTORY)
```

Comment:
Maybe we can remove the `PLATFORM_` prefix. What do you think?
On the other hand, now we use lowercase name with dash prefix for internal variables.

Reply:
`PLATFORM_` prefix removed.
Where can I find a lowercase name example along this file?